server/tests/backends/sql_translator/test_snowflake_translator_steps.py

```diff
@@ -91,62 +94,131 @@ def snowflake_query_executor(domain: str, query_string: str = None) -> Union[Tes
         return TestDataSlice(df=pd.DataFrame(res))
 
 
+def _drop_table(table_array: list):
+    global SNOWFLAKE_TABLES_TESTS
+
+    """
+    This method will drop tables
+    As given parameter, we have list of tables we want to drop
+
+    """
+    print("[x] Cleaning tables in progress, please wait...")
+    # Drop created table
+    for case_id in table_array:
+        # The try catch just if we're droping an already drope table in the process
+        try:
+            print(f"[-] Droping {case_id}")
+            execute(get_connection(), f'DROP TABLE {case_id};')
+            # we remove from SNOWFLAKE_TABLES_TESTS the case test
+            SNOWFLAKE_TABLES_TESTS.remove(case_id) if case_id in SNOWFLAKE_TABLES_TESTS else print(
+                f"{case_id} not in the list anymore !"
+            )
+        except Exception as es:
+            print(es)
+            print(f"[-] {case_id} was probably already dropped or not available !")
+
+
+def clean_too_old_residuals_tables():
+    """
+    This function  will get the list of all available tables
+    and then clean the too much old one by calculate the time of the creation of those tables
+
+    """
+    try:
+        # we get table list from snowflake
+        tables_list: pd.DataFrame = execute(get_connection(), "SHOW TABLES;")
+        too_old_tables_to_delete: list = []
+
+        # we loop on table's name of the database
+        for table_name in tables_list["name"].to_list():
+            if "_TOUCAN_TEST_" in table_name:
+                created_table_time = int(table_name.split("___")[1])
+                # if the table is too old (Was created 24 hour earlier)
+                if time.time() - created_table_time > 86400:
+                    too_old_tables_to_delete.append(table_name)
+
+        # then we delete the list of residual tables:
+        _drop_table(too_old_tables_to_delete)
+    except Exception as es:
+        # in some case we can have an exception here for an empty result for table
+        # no worries, not a big deal, that's why there is this error skip here
+        print(es)
+
+
 # Translation from Pipeline json to SQL query
 @pytest.mark.parametrize('case_id, case_spec_file_path', test_cases)
 def test_sql_translator_pipeline(case_id, case_spec_file_path, get_engine):
-    # TO prevent conflict on snowflake tables when testing on multiple terminals
-    case_id = case_id.replace("/", "") + str(randint(1, 100000))
+    global SNOWFLAKE_TABLES_TESTS
+
+    # Let's clean some old residuals tables
+    clean_too_old_residuals_tables()
```

Comment:
Thanks for this cleaning function ! 
Let's see if it doesn't slow too much the CI tests. 
If it's the case, perhaps we should try to do the same thing directly with a scheduled job in Snowflake.

Reply:
Updated !
Noticed, the actual function is been called multiple time in a same session tests
To fix that, i added a boolean checker that will allow the function to be executed only when it is False.